homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,486 @@
+                else:
+                    networks[i] = ssid
+
+        for net in networks:
+            hass.add_job(
+                async_add_entities,
+                [FritzBoxWifiSwitch(fritzbox_tools, entry.title, net, networks[net])],
+            )
+
+    hass.async_add_executor_job(_create_wifi_switches)
+    hass.async_add_executor_job(_create_port_switches)
+    hass.async_add_executor_job(_create_deflection_switches)
+    hass.async_add_executor_job(_create_profile_switches)
+
+
+class FritzBoxPortSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools PortForward switch."""
+
+    def __init__(
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name,
+        port_mapping,
+        idx,
+        connection_type,
+    ):
+        """Init Fritzbox port switch."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._attributes = {}
+        self.connection_type = connection_type
+        self.port_mapping: dict = port_mapping  # dict in the format as it comes from fritzconnection. eg: {'NewRemoteHost': '0.0.0.0', 'NewExternalPort': 22, 'NewProtocol': 'TCP', 'NewInternalPort': 22, 'NewInternalClient': '192.168.178.31', 'NewEnabled': True, 'NewPortMappingDescription': 'Beast SSH ', 'NewLeaseDuration': 0}
+        self._idx = idx  # needed for update routine
+
+        switch_info = SwitchInfo(
+            description=f'Port forward {port_mapping["NewPortMappingDescription"]}',
+            friendly_name=device_friendly_name,
+            icon="mdi:check-network",
+            type=SWITCH_TYPE_PORTFORWARD,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_handle_port_switch_on_off,
+        )
+        super().__init__(fritzbox_tools, switch_info)
+
+    async def _async_fetch_update(self):
+        """Fetch updates."""
+
+        try:
+            self.port_mapping = await self.hass.async_add_executor_job(
+                lambda: self.fritzbox_tools.connection.call_action(
+                    self.connection_type,
+                    "GetGenericPortMappingEntry",
+                    NewPortMappingIndex=self._idx,
+                )
+            )
+            _LOGGER.debug(
+                "Specific %s response: %s", SWITCH_TYPE_PORTFORWARD, self.port_mapping
+            )
+            self._attr_is_on = self.port_mapping["NewEnabled"] is True
+            self._is_available = True
+
+            self._attributes["internalIP"] = self.port_mapping["NewInternalClient"]
+            self._attributes["internalPort"] = self.port_mapping["NewInternalPort"]
+            self._attributes["externalPort"] = self.port_mapping["NewExternalPort"]
+            self._attributes["protocol"] = self.port_mapping["NewProtocol"]
+            self._attributes["description"] = self.port_mapping[
+                "NewPortMappingDescription"
+            ]
```

Comment:
Make a mapping dict to convert the key names

Reply:
Done